core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -473,12 +473,55 @@ class GroupCoordinator(val brokerId: Int,
     }
   }
 
+  private def onGroupUnloaded(groupEmigrating: Boolean)(group: GroupMetadata) {
```

Comment:
I slightly favor keeping them here since that keeps all state transitions and response callback handling confined to GroupCoordinator, but I don't have a strong preference if that is easier to understand.